### ISSUE TITLE
TypeError: boolean value of NA is ambiguous while running  describe_df(df)

### ISSUE THREAD

**Issue by zkid18** [pandas-profiling/pandas-profiling#437]
**Describe the bug**

variables:   9%|████████████████▊                                                                                                                                                                            | 8/90 [01:27<15:01, 10.99s/it, feature_name=my_numerical_feature_name]

Error traceback:

```
  File "create_report.py", line 56, in create_eda_html
    profile = ProfileReport(df_to_profile, title='Pandas Profiling Report', html={'style': {'full_width': True}})
  File "/report_env/src/pandas-profiling/src/pandas_profiling/__init__.py", line 63, in __init__
    description_set = describe_df(df)
  File "/report_env/src/pandas-profiling/src/pandas_profiling/model/describe.py", line 596, in describe
    executor.imap_unordered(multiprocess_1d, args)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/pool.py", line 761, in next
    raise value
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/report_env/src/pandas-profiling/src/pandas_profiling/model/describe.py", line 387, in multiprocess_1d
    return column, describe_1d(series)
  File "/report_env/src/pandas-profiling/src/pandas_profiling/model/describe.py", line 367, in describe_1d
    type_to_func[series_description["type"]](series, series_description)
  File "/report_env/src/pandas-profiling/src/pandas_profiling/model/describe.py", line 68, in describe_numeric_1d
    "n_zeros": (len(series) - np.count_nonzero(series)),
  File "<__array_function__ internals>", line 6, in count_nonzero
  File "/report_env/lib/python3.6/site-packages/numpy/core/numeric.py", line 452, in count_nonzero
    return multiarray.count_nonzero(a)
  File "pandas/_libs/missing.pyx", line 360, in pandas._libs.missing.NAType.__bool__
TypeError: boolean value of NA is ambiguous
```

**To Reproduce**
I tried to reproduce it, but the mocked seems working fine - no exceptions were raised.

```
def create_eda_html(report_file_path):
    df_to_profile = pd.DataFrame(data=np.random.randint(low=0, high=24, size=5627871))
    logging.info("Start profiling for df with shape {}".format(df_to_profile.shape))
    profile = ProfileReport(df_to_profile, title='Pandas Profiling Report', html={'style': {'full_width': True}})
    if os.path.exists(os.path.join(ROOT_DIR, report_file_path)) and report_file_path.endswith('.html'):
        profile.to_file(output_file=os.path.join(ROOT_DIR, report_file_path))
    else:
        profile.to_file(output_file="your_report.html")
    logging.info("Pandas profiling file created")


if __name__ == '__main__':
    parser = argparse.ArgumentParser(description='Read cofing')
    parser.add_argument('-c', '--file_path', default='report/reproduce.html', required=False,  help='Report sava path')
    args = parser.parse_args()
    create_eda_html(args.file_path)
```

Any advices about error reproduction are appreciated.

_Data:_
> train_df['my_numerical_feature_name'].describe()

```
count    5.627871e+06
mean     5.767664e+00
std      6.405506e+00
min      0.000000e+00
25%      0.000000e+00
50%      0.000000e+00
75%      1.200000e+01
max      2.400000e+01
Name: from_expdelv_deadline, dtype: float64
```

>np.count_nonzero(train_df['my_numerical_feature_name'])
```
2580030
```

>train_df['my_numerical_feature_name'].isna().sum()
```
0
```

_Code:_ Preferably, use this code format:
```
    df_to_profile = df
    logging.info("Start profiling for df with shape {}".format(df_to_profile.shape))
    profile = ProfileReport(df_to_profile, title='Pandas Profiling Report', html={'style': {'full_width': True}})
    if os.path.exists(os.path.join(ROOT_DIR, report_file_path)) and report_file_path.endswith('.html'):
        profile.to_file(output_file=os.path.join(ROOT_DIR, report_file_path))
    else:
        profile.to_file(output_file="your_report.html")
    logging.info("Pandas profiling file created")
```
--> 

**Version information:**

Version information is essential in reproducing and resolving bugs. Please report:

* _Python version_: Python 3.6.7
* _Environment_: command line
* _`pip`_: 

<details><summary>Click to expand <strong><em>Version information</em></strong></summary>
<p>

```
appnope==0.1.0
astroid==2.3.3
astropy==4.0.1.post1
attr==0.3.1
attrs==19.3.0
backcall==0.1.0
beautifulsoup4==4.8.2
bleach==3.1.4
bs4==0.0.1
certifi==2019.11.28
chardet==3.0.4
confuse==1.0.0
cycler==0.10.0
decorator==4.4.2
defusedxml==0.6.0
entrypoints==0.3
et-xmlfile==1.0.1
flake8==3.7.9
htmlmin==0.1.12
idna==2.8
importlib-metadata==1.6.0
ipykernel==5.2.0
ipython==7.13.0
ipython-genutils==0.2.0
ipywidgets==7.5.1
isort==4.3.21
jedi==0.16.0
Jinja2==2.11.1
joblib==0.14.1
jsonschema==3.2.0
jupyter-client==5.3.4
jupyter-core==4.6.1
kaggle==1.5.6
kiwisolver==1.2.0
lazy-object-proxy==1.4.3
llvmlite==0.31.0
lxml==4.5.0
MarkupSafe==1.1.1
matplotlib==3.2.1
mccabe==0.6.1
missingno==0.4.2
mistune==0.8.4
more-itertools==8.2.0
nbconvert==5.6.1
nbformat==5.0.5
networkx==2.4
notebook==6.0.3
numba==0.48.0
numpy==1.18.2
packaging==20.3
pandas==0.25.3
pandas-profiling==2.5.0
pandocfilters==1.4.2
parso==0.6.2
pexpect==4.8.0
phik==0.9.9
pickleshare==0.7.5
pluggy==0.13.1
prometheus-client==0.7.1
prompt-toolkit==3.0.5
ptyprocess==0.6.0
py==1.8.1
pycodestyle==2.5.0
pyflakes==2.1.1
Pygments==2.6.1
pylint==2.4.4
pyparsing==2.4.7
pyrsistent==0.16.0
pytest==5.4.1
pytest-pylint==0.15.1
python-dateutil==2.8.1
python-slugify==4.0.0
pytz==2019.3
PyYAML==5.3
pyzmq==19.0.0
requests==2.22.0
scikit-learn==0.22.2.post1
scipy==1.4.1
seaborn==0.10.0
Send2Trash==1.5.0
six==1.14.0
sklearn==0.0
soupsieve==2.0
tangled-up-in-unicode==0.0.3
terminado==0.8.3
testpath==0.4.4
text-unidecode==1.3
tornado==6.0.4
tqdm==4.42.0
traitlets==4.3.3
typed-ast==1.4.1
urllib3==1.24.3
visions==0.2.2
wcwidth==0.1.9
webencodings==0.5.1
websocket-client==0.57.0
widgetsnbextension==3.5.1
wrapt==1.11.2
zipp==3.1.0
```
</p>
</details>

**Additional context**

**Comment by loopyme** [comment]
That's a bug in `pandas_profiling.model.describe.describe_numeric_1d` function (or in my PR:``pandas_profiling.model.statistic.describe_numeric_1d` function)

In this function, `numpy.count_nonzero()` is called with a `pandas.Series` as input, which is slowly and risky especially when series contains Na.

I was planning to optimize some low-level functions to speed things up and make PP more stable. In fact the bug you mentioned has been fixed in my local branch, so I can commit the patch and add issue test later in my next PR.

(Wow, I've written a lot of code in the last few days. I am now stall and waiting for review.)

**Comment by zkid18** [comment]
Thanks for the reply. Probably need to report the bug to numpy?
Have you find out what causes the riskiness while calling `numpy.count_nonzero()`  with a `pandas.Series`? Also in my example, there are no missing values in the series.

**Comment by loopyme** [comment]
I think it's `pd.NA` that causes this bug and bring riskiness to this method, and `np.count_nonzero(pd.Series([pd.NA]))` will reproduce the bug.

I didn't figure out if this is a bug in the way pd passed values to np, or a bug in `np.count_nonzero`, or bug in `pd.NA` itself, so I haven't reported this bug yet.

According to your error trace back, It's definitely `pd.NA`(`pandas._libs.missing.NA`) that causes the bug. However, since I can't test on your data, I don't know why it's in your data frame.

**Comment by sbrugman** [comment]
Thanks to @loopyme, this will be resolved in v2.7.0.
